src/android/com/microsoftopentechnologies/intellij/wizards/createvm/EndpointStep.java

```diff
@@ -0,0 +1,396 @@
+
+                    virtualMachine.getEndpoints().addAll(tableModel.getData());
+
+                    AzureSDKManagerImpl.getManager().createVirtualMachine(virtualMachine, model.getVirtualMachineImage(), model.getStorageAccount(), model.getUserName(), new String(model.getPassword()));
+
+                    ApplicationManager.getApplication().invokeLater(new Runnable() {
+                        @Override
+                        public void run() {
+                            try {
+                                node.addChildNode(new VMNode(node, virtualMachine));
+                            } catch (AzureCmdException e) {
+                                UIHelper.showException("Error refreshing VM list", e);
+                            }
+                        }
+                    });
+                } catch (AzureCmdException e) {
+                    UIHelper.showException("Error creating virtual machine", e);
+                }
+            }
+        });
+
+        return super.onFinish();
+    }
+
+    private void createUIComponents() {
+        endpointsTable = new JBTable() {
+            @Override
+            public TableCellRenderer getCellRenderer(int row, int col) {
+                switch(col){
+                    case 0:
+                        return new ErrorRenderer();
+                    case 4:
+                        return new ProtocolRenderer();
+                    case 5:
+                        return new DeleteRenderer();
+                    default:
+                        return super.getCellRenderer(row, col);
+                }
+            }
+
+            @Override
+            public TableCellEditor getCellEditor(int row, int col) {
+                if(col == 4) {
+                    return new DefaultCellEditor(new ComboBox(new String[] { "TCP", "UDP" }));
+                } else {
+                    return super.getCellEditor(row, col);
+                }
+            }
+        };
+    }
+
+    private static String getErrorFromRow(int row, Vector<Endpoint> list) {
+        String errors = "";
+        Endpoint endpoint = list.get(row);
+
+        if (endpoint.getName().length() < 3 && endpoint.getName().length() > 15) {
+            errors = errors + "The name must between 3 and 15 character long. \n";
+        }
+
+        if (!endpoint.getName().matches("^[A-Za-z0-9][A-Za-z0-9-\\s]+[A-Za-z0-9]$")) {
+            errors = errors + "The name must start with a letter or number, " +
+                    "contain only letters, numbers, and hyphens, " +
+                    "and end with a letter or number. \n";
+        }
+
+
+        if (endpoint.getPrivatePort() < 1 || endpoint.getPrivatePort() > 65535) {
+            errors = errors + "The private port must between a number between 1 and 65535. \n";
+        }
+
+        boolean containsName = false;
+        for (Endpoint ep : list) {
+            if(ep != endpoint && ep.getName().equals(endpoint.getName())) {
```

Comment:
This is a case sensitive compare. Does Azure allow endpoint names to differ only in case?